platform/posix/transport/utest/CMakeLists.txt

```diff
@@ -152,3 +152,33 @@ create_test(${utest_name}
            "${utest_dep_list}"
            "${test_include_directories}"
         )
+
+# list the files you would like to test here
+set(real_source_files
+        ${RECONNECT_SOURCES}
+        )
+set(real_name "transport_reconnect_real")
+
+create_real_library(${real_name}
+                    "${real_source_files}"
+                    "${real_include_directories}"
+                    "${mock_name}"
```

Comment:
This will include all mock files defined at the top of the file.  It seems like this file has become a mish-mash of multiple test targets. 
As the reconnect utilities may be possibly moved outside of transport (as it has no direct relation to transport layer but is purely a math retry utility), it would be better to separately define its test configuration with the mock dependencies directly related to it